crates/sui-types/src/event.rs

```diff
@@ -73,33 +75,110 @@ pub enum TransferType {
     EnumDiscriminants,
     EnumVariantNames,
 )]
-#[strum_discriminants(name(EventType))]
+#[strum_discriminants(name(EventType), derive(Serialize, Deserialize, JsonSchema))]
 // Developer note: PLEASE only append new entries, do not modify existing entries (binary compat)
 pub enum Event {
     /// Move-specific event
-    MoveEvent(MoveObject),
+    MoveEvent {
+        package_id: ObjectID,
+        module: Identifier,
+        function: Identifier,
```

Comment:
For these module/function pairs. Could you make it something like `transaction_module` and `transaction_function`?
Or something similar.

My concern is it is not obvious what these are doing here. At first I thought these were somehow related to the `type_`, not the transaction that created the event.